numpy/lib/function_base.py

```diff
@@ -1323,25 +1323,24 @@ def piecewise(x, condlist, funclist, *args, **kw):
     """
     x = asanyarray(x)
     n2 = len(funclist)
-    if (isscalar(condlist) or not (isinstance(condlist[0], list) or
-                                   isinstance(condlist[0], ndarray))):
-        if not isscalar(condlist) and x.size == 1 and x.ndim == 0:
-            condlist = [[c] for c in condlist]
-        else:
-            condlist = [condlist]
+
+    # undocumented: single condition is promoted to a list of one condition
+    if isscalar(condlist) or (
+            not isinstance(condlist[0], (list, ndarray)) and x.ndim != 0):
+        condlist = [condlist]
+
```

Comment:
So, was the idea to deprecate this? An alternative might be to just add `ndmin=1` in the conversion to boolean array below.

Reply:
I'd like to deprecate this in future, but this patch is orthogonal to that

Reply:
`ndim=1` won't help - this is intended to also promote 1d arrays to a list of length 1 containing a single 1d array.